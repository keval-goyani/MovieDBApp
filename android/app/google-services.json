{
  "project_info": {
    "project_number": "89996564819",
    "project_id": "moviedb-2908d",
    "storage_bucket": "moviedb-2908d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:89996564819:android:e942630a87714b1d3ea9a6",
        "android_client_info": {
          "package_name": "com.moviedbapp"
        }
      },
      "oauth_client": [
        {
          "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBYrM1MC9Go35l8ra748zqBtCRDkrB_Nn4"
        }
      ],
      "services": {
        "appinvite_service": {
          "other_platform_oauth_client": [
            {
              "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
              "client_type": 3
            }
          ]
        }
      }
    }
  ],
  "configuration_version": "1"
}