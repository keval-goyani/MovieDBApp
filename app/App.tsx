import React, {useEffect} from 'react';
import {SafeAreaView, Text} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {styles} from './theme';

const App = () => {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Text>Hello World</Text>
    </SafeAreaView>
  );
};

export default App;
