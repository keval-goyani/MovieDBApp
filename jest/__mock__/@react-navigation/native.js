export const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  replace: jest.fn(),
  popToTop: jest.fn(),
  addListener: jest.fn().mockImplementation((str, cb) => cb()),
  pop: jest.fn(),
  getParam: jest.fn(),
  push: jest.fn(),
};

const useRoute = jest.fn(val => val);
// const useFocusEffect = jest.fn();
const useIsFocused = jest.fn(val => val);
const useNavigation = () => navigation;
const setFocus = jest.fn();
const createNavigationContainerRef = jest.fn();
const DefaultTheme = {};

const CommonActions = { reset: jest.fn() };
const StackActions = {
  pop: jest.fn(),
};

export {
  useRoute,
  useNavigation,
  // useFocusEffect,
  useIsFocused,
  setFocus,
  createNavigationContainerRef,
  DefaultTheme,
  CommonActions,
  StackActions,
};
