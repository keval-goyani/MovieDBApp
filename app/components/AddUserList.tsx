import { useFocusEffect, useNavigation } from '@react-navigation/native';
import React, { useEffect, useState } from 'react';
import { FlatList, Text, TouchableOpacity, View } from 'react-native';
import Check from 'react-native-vector-icons/AntDesign';
import { useDispatch, useSelector } from 'react-redux';
import { asMutable } from 'seamless-immutable';
import { ProfileImage, UserListEmpty } from '../components';
import {
  AddUserListProps,
  NavigationDataType,
  navigationStrings,
  RenderItemTypes,
  SelectedUsersProps,
  strings,
  UserDataType,
} from '../constants';
import { authDataSelectors } from '../redux/AuthRedux';
import { userListSelector } from '../redux/UserListRedux';
import { conversationIdCreation, sub } from '../services';
import { Color, moderateScale } from '../theme';
import { styles } from './styles/AddUserListStyles';

const AddUsersList = ({ userListData, setSelectedUsers }: AddUserListProps) => {
  const navigation: NavigationDataType = useNavigation();
  const dispatch = useDispatch();
  const { user } = useSelector(authDataSelectors.getData);
  const { fetchingUserList } = useSelector(userListSelector.getData);
  const [selectedItems, setSelectedItems] = useState<UserDataType[]>([]);

  useFocusEffect(() => {
    sub({ user, dispatch });
    return () => sub;
  });

  useEffect(() => {
    setSelectedUsers(selectedItems);
  }, [selectedItems, setSelectedUsers]);

  const renderUserList = ({ item, index }: RenderItemTypes) => {
    const usersEmail = [user?.email ?? '', item?.email];
    const conversationId: string = conversationIdCreation(usersEmail);

    const handleOnPress = (selectedUser: UserDataType) => {
      console.log(selectedItems, '<==selectedItems.length');
      console.log(selectedUser, '<==selectedUser');

      if (selectedItems.length) {
        return selectUsers(selectedUser);
      }
      navigation.goBack();
      navigation.navigate(navigationStrings.Chat, {
        conversationId,
        username: item?.username,
        receiverId: item?.uid,
        userStatus: item?.status,
        profileImage: item?.profileImage,
        userEmail: item?.email,
      });
    };

    const getSelected = (usersData: SelectedUsersProps) => {
      return JSON.stringify(selectedItems).includes(JSON.stringify(usersData));
    };

    const selected = getSelected(item);

    const selectUsers = (selectedUser: UserDataType) => {
      if (
        JSON.stringify(selectedItems).includes(JSON.stringify(selectedUser))
      ) {
        const filteredList = selectedItems?.filter(
          (items: UserDataType) => items !== selectedUser,
        );
        setSelectedItems(filteredList);
      } else {
        setSelectedItems([...selectedItems, selectedUser]);
      }
    };

    return (
      <TouchableOpacity
        style={styles.listItem}
        onPress={() => handleOnPress(item)}x
        activeOpacity={0.8}
        onLongPress={() => selectUsers(item)}
        testID={`button-for-test${index}`}>
        <View style={styles.itemContainer}>
          <ProfileImage
            profileImage={item?.profileImage}
            userStatus={item?.status}
          />
          <View style={styles.nameView}>
            <Text style={styles.text}>{item?.username}</Text>
            <Text
              style={styles.lastChatText}
              ellipsizeMode="tail"
              numberOfLines={1}>
              {item?.email}
            </Text>
          </View>
        </View>
        {selected && (
          <>
            <View style={styles.overlay} />
            <Check
              testID="check"
              name={strings.checkIcon}
              size={moderateScale(23)}
              color={Color.lightBlue}
              style={styles.iconStyle}
            />
          </>
        )}
      </TouchableOpacity>
    );
  };

  return (
    <FlatList
      testID={'flatList'}
      data={asMutable(userListData)}
      renderItem={renderUserList}
      ListEmptyComponent={
        <UserListEmpty
          fetching={fetchingUserList}
          userListLength={userListData.length}
        />
      }
      showsVerticalScrollIndicator={false}
      bounces={false}
    />
  );
};

export default AddUsersList;
