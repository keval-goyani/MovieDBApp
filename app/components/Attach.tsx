import { useNavigation } from '@react-navigation/native';
import React, { FC } from 'react';
import { Alert, View } from 'react-native';
import SystemSetting from 'react-native-system-setting';
import { CustomIconRounder } from '../components';
import {
  AttachDataType,
  NavigationDataType,
  navigationStrings,
  strings,
} from '../constants';
import {
  handleCameraPermission,
  handleDocumentPermission,
  handleGalleryPermission,
} from '../services';
import { Icons } from '../theme';
import styles from './styles/AttachStyles';

const Attach: FC<AttachDataType> = ({
  setIsAttach,
  setImagePath,
  setDocumentData,
  conversationId,
  username,
  receiverId,
}) => {
  const navigation: NavigationDataType = useNavigation();

  const turnOnLocation = (enable: boolean) => {
    if (enable) {
      navigation.navigate(navigationStrings.Location, {
        isFromChat: false,
        ...{ conversationId, username, receiverId },
      });
      setIsAttach(false);
    } else {
      Alert.alert(strings.turnOn, strings.goToSetting, [
        {
          onPress: () => SystemSetting.switchLocation(),
          text: strings.setting,
        },
        { text: strings.cancel },
      ]);
    }
  };

  const locationHandler = () => {
    SystemSetting.isLocationEnabled()
      .then(enable => {
        turnOnLocation(enable);
      })
      .catch(error => Alert.alert(strings.warning, error));
  };

  return (
    <View style={styles.container}>
      <View style={styles.iconContainer}>
        <CustomIconRounder
          backgroundColor={styles.documentBackgroundColor}
          path={Icons.fileIcon}
          iconName={strings.document}
          tintColor={styles.documentTintColor}
          onPress={() => {
            setIsAttach(false);
            handleDocumentPermission(setDocumentData);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.cameraBackgroundColor}
          path={Icons.cameraIcon}
          iconName={strings.camera}
          tintColor={styles.cameraTintColor}
          onPress={() => {
            handleCameraPermission(setImagePath);
            setIsAttach(false);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.galleryBackgroundColor}
          path={Icons.galleryIcon}
          iconName={strings.gallery}
          tintColor={styles.galleryTintColor}
          onPress={() => {
            handleGalleryPermission(setImagePath);
            setIsAttach(false);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.locationBackgroundColor}
          path={Icons.locationIcon}
          iconName={strings.map}
          tintColor={styles.locationTintColor}
          onPress={locationHandler}
        />
      </View>
    </View>
  );
};

export default Attach;
