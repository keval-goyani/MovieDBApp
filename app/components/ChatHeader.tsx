import { useNavigation } from '@react-navigation/native';
import React, { useState } from 'react';
import {
  Image,
  Pressable,
  PressableStateCallbackType,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { ChatMenu, ProfileImage } from '../components';
import {
  appConstants,
  ChatHeaderDataType,
  NavigationDataType,
  navigationStrings,
} from '../constants';
import { Color, Icons } from '../theme';
import { styles } from './styles/ChatHeaderStyles';

const ChatHeader = ({
  username,
  profileImage,
  userStatus,
  showMenu,
  setShowMenu,
  setIsAttach,
  setCameraModal,
  setChatWallpaper,
  conversationId,
  membersName,
  groupName,
  userEmail,
  receiverId,
}: ChatHeaderDataType) => {
  const navigation: NavigationDataType = useNavigation();
  const userInfo = userStatus ?? membersName;

  const styleFn = ({ pressed }: PressableStateCallbackType) => [
    {
      backgroundColor: pressed ? Color.transparentGrey : Color.darkBlue,
    },
    styles.userNameAndStatus,
  ];

  return (
    <>
      <View style={styles.container}>
        <TouchableOpacity
          onPress={() => navigation.goBack()}
          testID={'goBackButton'}>
          <Image source={Icons?.backIcon} style={styles.leftIconStyle} />
        </TouchableOpacity>
        <View style={styles.profileView}>
          <View style={styles.profile}>
            <ProfileImage
              {...{ profileImage, userStatus, groupName }}
              isChatHeader={appConstants.trueValue}
            />
            <Pressable
              testID={'profileImagePressable'}
              onPress={() =>
                navigation.navigate(navigationStrings.ProfileInfo, {
                  profileImage,
                  username,
                  userEmail,
                  conversationId,
                  receiverId,
                  groupName,
                })
              }
              style={styleFn}>
              <Text style={styles.username}>{username}</Text>
              <Text numberOfLines={1} style={styles.statusOrMembers}>
                {userInfo}
              </Text>
            </Pressable>
          </View>
        </View>
        <Pressable
          testID={'dotsMenu'}
          onPress={() => {
            setShowMenu(!showMenu);
            setIsAttach(false);
            setCameraModal(false);
          }}>
          <Image source={Icons?.dotMenu} style={styles.dotsMenu} />
        </Pressable>
      </View>
      {showMenu && (
        <ChatMenu {...{ conversationId, setChatWallpaper, setShowMenu }} />
      )}
    </>
  );
};

export default ChatHeader;
