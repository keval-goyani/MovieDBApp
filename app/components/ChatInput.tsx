import React, {
  FC,
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from 'react';
import {
  Image,
  Keyboard,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
import { useSelector } from 'react-redux';
import { Attach, Stagger } from '../components';
import { ChatInputDataType, strings } from '../constants';
import { authDataSelectors } from '../redux/AuthRedux';
import { chatCreation } from '../services';
import { Color, Icons } from '../theme';
import { styles } from './styles/ChatInputStyles';

const ChatInput: FC<ChatInputDataType> = ({
  conversationId,
  cameraModal,
  setCameraModal,
  isAttach,
  setIsAttach,
  setImagePath,
  documentData,
  setDocumentData,
  imageUrl,
  setShowMenu,
  username,
  receiverId,
  members,
}) => {
  const [message, setMessage] = useState<string>('');
  const { user } = useSelector(authDataSelectors.getData);
  const messageInput = useRef<TextInput>(null);
  const fixedMessage = useMemo(() => {
    return {
      conversationId,
      senderId: user?.uid ?? '',
      receiverId: receiverId ?? '',
      members,
    };
  }, [conversationId, members, receiverId, user]);

  const imageMessageCreation = useCallback(() => {
    const imageMessage = {
      ...fixedMessage,
      content: imageUrl,
      type: strings.imageType,
    };

    chatCreation(imageMessage);
  }, [fixedMessage, imageUrl]);

  const documentMessageCreation = useCallback(() => {
    const { documentName, documentUrl } = documentData;
    const documentMessage = {
      ...fixedMessage,
      content: documentUrl,
      type: strings.document,
      documentName,
    };

    chatCreation(documentMessage);
  }, [documentData, fixedMessage]);

  const textMessageCreation = useCallback(() => {
    const textMessage = {
      ...fixedMessage,
      content: message.trim(),
      type: strings.textMessageType,
    };

    chatCreation(textMessage);
  }, [fixedMessage, message]);

  const sendMessageHandler = useCallback(() => {
    if (message.trim().len gth !== 0) {
      textMessageCreation();
      setMessage('');
    }
  }, [textMessageCreation, message]);

  useEffect(() => {
    imageUrl && imageMessageCreation();
  }, [imageMessageCreation, imageUrl]);

  useEffect(() => {
    documentData?.documentUrl && documentMessageCreation();
  }, [documentMessageCreation, documentData]);

  useEffect(() => {
    messageInput && messageInput?.current?.focus();
  }, []);

  return (
    <View style={styles.container}>
      <View style={styles.innerContainer}>
        <View style={styles.inputAndSend}>
          <TextInput
            testID="textInput"
            multiline
            placeholder={strings.chatPlaceholder}
            placeholderTextColor={Color.darkGray}
            style={styles.input}
            onChangeText={text => setMessage(text)}
            value={message}
            onFocus={() => {
              setCameraModal(false);
              setIsAttach(false);
            }}
            ref={messageInput}
          />
          <TouchableOpacity
            testID="AttachIcon"
            onPress={() => {
              setIsAttach(!isAttach);
              setCameraModal(false);
              setShowMenu(false);
              Keyboard.dismiss();
            }}>
            <Image source={Icons?.attach} style={styles.inputIcon} />
          </TouchableOpacity>
          <TouchableOpacity
            testID="CameraIcon"
            onPress={() => {
              setCameraModal(!cameraModal);
              setIsAttach(false);
              setShowMenu(false);
              Keyboard.dismiss();
            }}>
            <Image source={Icons?.camera} style={styles.inputIcon} />
          </TouchableOpacity>
        </View>
        <TouchableOpacity
          testID="sendButton"
          style={styles.sendButtonView}
          onPress={() => {
            sendMessageHandler();
            setCameraModal(false);
            setIsAttach(false);
            setShowMenu(false);
          }}>
          <Image source={Icons?.send} style={styles.sendButton} />
        </TouchableOpacity>
      </View>
      {cameraModal && (
        <Stagger {...{ cameraModal, setCameraModal, setImagePath }} />
      )}
      {isAttach && (
        <Attach
          {...{
            setIsAttach,
            setImagePath,
            setDocumentData,
            conversationId,
            username,
            receiverId,
          }}
        />
      )}
    </View>
  );
};

export default ChatInput;
