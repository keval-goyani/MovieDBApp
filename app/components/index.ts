import ChatHeader from './ChatHeader';
import ChatInput from './ChatInput';
import CustomSkeleton from './CustomSkeleton';
import DropDownMenu from './DropDownMenu';
import Header from './Header';
import List from './List';
import ListContainer, { movieListItem } from './ListContainer';
import Loader from './Loader';
import LoadingState from './LoadingState';
import Message from './Message';
import MessageList from './MessageList';
import Movies from './Movies';
import MovieTrailer, { trailerListItem } from './MovieTrailers';
import SearchModal from './SearchModal';
import SkeletonCard from './SkeletonCard';
import TabBarIcon from './TabBarIcon';
import Trailers from './Trailers';
import UserListEmpty from './UserListEmpty';
import UsersList from './UsersList';

export {
  ChatHeader,
  ChatInput,
  DropDownMenu,
  Header,
  List,
  ListContainer,
  MovieTrailer,
  Message,
  MessageList,
  TabBarIcon,
  SearchModal,
  Loader,
  LoadingState,
  Movies,
  movieListItem,
  Trailers,
  trailerListItem,
  UsersList,
  SkeletonCard,
  CustomSkeleton,
  UserListEmpty,
};
