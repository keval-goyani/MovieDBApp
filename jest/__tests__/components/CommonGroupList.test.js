import { navigation } from '@react-navigation/native';
import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { CommonGroupList } from '../../../app/components';

const listOfGroups = [{ groupName: 'RNTeam' }];
const listOfGroups2 = [
  {
    groupName: 'RNTeam',
    conversationId: 'conversationId1',
    profileImage: 'groupImage1',
  },
  {
    groupName: 'MyTeam',
    conversationId: 'conversationId2',
    profileImage: 'groupImage2',
  },
];

const listWithMembers = [{ members: ['chandani', 'keval'] }];

describe('common Group List', () => {
  it('should take snapShot of CommonGroupList component', () => {
    const { toJSON } = render(<CommonGroupList groupList={listOfGroups} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the list correctly ', () => {
    const { getByTestId, getAllByTestId } = render(
      <CommonGroupList groupList={listOfGroups} />,
    );

    const list = getByTestId('Common-Group-List-');
    expect(list).toBeDefined();

    const buttons = getAllByTestId(/button/);
    expect(buttons).toHaveLength(listOfGroups.length);
  });

  it('should correctly generate membersList array', () => {
    const { getByTestId } = render(
      <CommonGroupList groupList={listWithMembers} />,
    );

    const button = getByTestId('button0');
    fireEvent.press(button);
    const { conversationId, groupName, username, profileImage, ...rest } =
      navigation.navigate.mock.calls[0][1];
    console.log(navigation.navigate.mock.calls, '<==navigation.navigate');
    console.log(rest);
    const membersList = Object.values(rest);
    expect(membersList).toEqual(['chandani', 'keval']);
  });

  it('should navigate to chat screen on group item press', () => {
    const { getByTestId } = render(
      <CommonGroupList groupList={listOfGroups2} />,
    );

    const button = getByTestId('button1');
    fireEvent.press(button);
    expect(navigation.navigate).toHaveBeenCalledWith('Chat', {
      conversationId: listOfGroups2[1].conversationId,
      groupName: listOfGroups2[1].groupName,
      username: listOfGroups2[1].groupName,
      profileImage: listOfGroups2[1].profileImage,
    });
  });
});
