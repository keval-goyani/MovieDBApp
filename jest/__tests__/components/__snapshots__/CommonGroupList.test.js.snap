// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common Group List should take snapShot of CommonGroupList component 1`] = `
<RCTScrollView
  bounces={false}
  data={
    Array [
      Object {
        "groupName": "RNTeam",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  testID="Common-Group-List-"
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": undefined,
            "borderColor": "#BBCFD5",
            "flexDirection": "row",
            "marginBottom": undefined,
            "opacity": 1,
            "overflow": "hidden",
          }
        }
        testID="button0"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "padding": undefined,
            }
          }
          testID="Common-Group-List"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0d253f",
                "borderColor": "#01b4e4",
                "borderRadius": 30,
                "borderWidth": undefined,
                "height": undefined,
                "justifyContent": "center",
                "width": undefined,
              }
            }
            testID="profile-image"
          >
            <Image
              source={
                Object {
                  "default": Object {
                    "attach": Object {
                      "testUri": "../../../app/assets/icons/attach.png",
                    },
                    "avatar": Object {
                      "testUri": "../../../app/assets/icons/user.png",
                    },
                    "backIcon": Object {
                      "testUri": "../../../app/assets/icons/backIcon.png",
                    },
                    "camera": Object {
                      "testUri": "../../../app/assets/icons/camera.png",
                    },
                    "cameraIcon": Object {
                      "testUri": "../../../app/assets/icons/cameraIcon.png",
                    },
                    "chatBackground": Object {
                      "testUri": "../../../app/assets/icons/chatBackground.png",
                    },
                    "close": Object {
                      "testUri": "../../../app/assets/icons/close.png",
                    },
                    "comingSoon": Object {
                      "testUri": "../../../app/assets/icons/comingSoon.png",
                    },
                    "communityIcon": Object {
                      "testUri": "../../../app/assets/icons/communityIcon.png",
                    },
                    "document": Object {
                      "testUri": "../../../app/assets/icons/documentIcon.png",
                    },
                    "documentIcon": Object {
                      "testUri": "../../../app/assets/icons/documentIcon.png",
                    },
                    "dotMenu": Object {
                      "testUri": "../../../app/assets/icons/dots.png",
                    },
                    "expandIcon": Object {
                      "testUri": "../../../app/assets/icons/expandIcon.png",
                    },
                    "fileIcon": Object {
                      "testUri": "../../../app/assets/icons/fileIcon.png",
                    },
                    "fullStop": Object {
                      "testUri": "../../../app/assets/icons/fullStop.png",
                    },
                    "galleryIcon": Object {
                      "testUri": "../../../app/assets/icons/galleryIcon.png",
                    },
                    "homeIcon": Object {
                      "testUri": "../../../app/assets/icons/homeIcon.png",
                    },
                    "imageDocument": Object {
                      "testUri": "../../../app/assets/icons/imageIcon.png",
                    },
                    "location": Object {
                      "testUri": "../../../app/assets/icons/share.png",
                    },
                    "locationIcon": Object {
                      "testUri": "../../../app/assets/icons/locationIcon.png",
                    },
                    "logout": Object {
                      "testUri": "../../../app/assets/icons/logout.png",
                    },
                    "menuIcon": Object {
                      "testUri": "../../../app/assets/icons/menuIcon.png",
                    },
                    "movieDbIcon": Object {
                      "testUri": "../../../app/assets/icons/movieDBIcon.png",
                    },
                    "movieDbIcon2x": Object {
                      "testUri": "../../../app/assets/icons/movieDB2x.png",
                    },
                    "movieDbIcon3x": Object {
                      "testUri": "../../../app/assets/icons/movieDB3x.png",
                    },
                    "moviesIcon": Object {
                      "testUri": "../../../app/assets/icons/moviesIcon.png",
                    },
                    "noItemFound": Object {
                      "testUri": "../../../app/assets/icons/noItemFound.png",
                    },
                    "notFound": Object {
                      "testUri": "../../../app/assets/icons/notFoundIcon.png",
                    },
                    "pdfDocument": Object {
                      "testUri": "../../../app/assets/icons/pdfDocument 2.png",
                    },
                    "playIcon": Object {
                      "testUri": "../../../app/assets/icons/playIcon.png",
                    },
                    "plus": Object {
                      "testUri": "../../../app/assets/icons/plusSign.png",
                    },
                    "pptDocument": Object {
                      "testUri": "../../../app/assets/icons/pptDocument.png",
                    },
                    "searchIcon": Object {
                      "testUri": "../../../app/assets/icons/searchIcon.png",
                    },
                    "send": Object {
                      "testUri": "../../../app/assets/icons/send.png",
                    },
                    "threeDotIcon": Object {
                      "testUri": "../../../app/assets/icons/threeDotIcon.png",
                    },
                    "trailerIcon": Object {
                      "testUri": "../../../app/assets/icons/trailerIcon.png",
                    },
                    "trending": Object {
                      "testUri": "../../../app/assets/icons/trending.png",
                    },
                    "tv": Object {
                      "testUri": "../../../app/assets/icons/tv.png",
                    },
                    "videoDocument": Object {
                      "testUri": "../../../app/assets/icons/mp4Document.png",
                    },
                  },
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 30,
                    "height": undefined,
                    "tintColor": "#01b4e4",
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "marginLeft": undefined,
                "marginTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontSize": undefined,
                  "fontWeight": "600",
                  "marginLeft": undefined,
                  "paddingBottom": undefined,
                  "textAlign": "justify",
                }
              }
            >
              RNTeam
            </Text>
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "color": "#000000",
                  "fontSize": undefined,
                  "marginLeft": undefined,
                  "paddingBottom": undefined,
                  "textAlign": "justify",
                }
              }
            >
              ,
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
